Fix for 22177995 /dev/console and /dev/msglog redirection getting garbled in legacy locale on s11

--- luit-20230201/luit.c
+++ luit-20230201/luit.c
@@ -26,6 +26,7 @@ */
 #include <luit.h>
 
 #include <locale.h>
+#include <sys/strredir.h>
 #include <fcntl.h>
 #include <unistd.h>
 #include <errno.h>
@@ -67,6 +68,7 @@ #endif
 
 static volatile int sigwinch_queued = 0;
 static volatile int sigchld_queued = 0;
+int Console = 0;
 
 static int convert(int, int);
 static int condom(int, char **);
@@ -483,6 +485,9 @@ } else if (!strcmp(argv[i], "-encoding")
 	    locale_name = getParam(i);
 	    ignore_locale = 1;
 	    i += 2;
+	} else if (!strcmp(argv[i], "-C")) {
+	    Console = 1;
+	    i += 1;
 	} else if (!strcmp(argv[i], "-p")) {
 	    pipe_option = 1;
 	    i += 1;
@@ -808,6 +813,14 @@ read_waitpipe(p2c_waitpipe);
 	close_waitpipe(0);
     }
 
+    if (Console) {
+	int consfd = open("/dev/console", O_RDONLY);
+	    if (consfd >= 0) {
+		 if (ioctl(consfd, SRIOCSREDIR, 0) != -1)
+		    close(consfd);
+	    }
+    }
+
     execvp(path, argv);
     perror(path);
     ExitFailure();
